lib/widget/widget.cpp

```diff
@@ -192,6 +246,7 @@ WIDGET::WIDGET(W_INIT const *init, WIDGET_TYPE type)
 
 WIDGET::WIDGET(WIDGET *parent, WIDGET_TYPE type): WIDGET(type)
 {
+	ASSERT_OR_RETURN(, parent, "WIDGET::WIDGET - parent is null!!");
```

Comment:
This constructor was removed in the widget smart pointer refactor. All widgets can be instantiated without parents in that branch. We could discuss this, but I would really like to keep the possibility of instantiating without parent. Most widgets had a "parent-less" constructor anyway (the one that accepts a `W_INIT` or some subclass).

Reply:
Yup that's fine - I figured this (and a few other changes) were going to go away once the widget smart pointer refactor is merged (first). 👍